packages/components/src/DataList/DataList.utils.tsx

```diff
@@ -137,3 +163,55 @@ export function generateDataListEmptyState({
 
   return EmptyStateComponent;
 }
+
+export function sortSizeProp(sizeProp: Breakpoints[]) {
+  return sizeProp.sort(
+    (a, b) => BREAKPOINTS.indexOf(a) - BREAKPOINTS.indexOf(b),
+  );
+}
+
+export function useGridLayoutMediaQueries() {
+  const mediaQueries = BREAKPOINTS.reduce(
+    (previous, breakpoint) => ({
+      ...previous,
+      [breakpoint]: window.matchMedia(breakpointToMediaQuery(breakpoint)),
+    }),
+    {} as Record<Breakpoints, MediaQueryList>,
+  );
+
+  const initialMatches = BREAKPOINTS.reduce(
+    (previous, breakpoint) => ({
+      ...previous,
+      [breakpoint]: mediaQueries[breakpoint].matches,
+    }),
+    {} as Record<Breakpoints, boolean>,
+  );
+
+  const [matches, setMatches] = useState(initialMatches);
+  // Set up mediaQuery event handlers
+  useEffect(() => {
+    const handlers = BREAKPOINTS.reduce((previous, breakpoint) => {
+      const handler = (e: MediaQueryListEvent) =>
+        setMatches(previousMatches => ({
+          ...previousMatches,
+          [breakpoint]: e.matches,
+        }));
+      mediaQueries[breakpoint].addEventListener("change", handler);
+      return { ...previous, [breakpoint]: handler };
+    }, {} as Record<Breakpoints, (e: MediaQueryListEvent) => void>);
+    return () => {
+      BREAKPOINTS.map(breakpoint => {
```

Comment:
semantically, this is a `forEach`, no? we aren't really using what `map` is returning